core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1191,6 +1227,27 @@ class KafkaController(val config: KafkaConfig,
     globalTopicCount = if (!isActive) 0 else controllerContext.allTopics.size
 
     globalPartitionCount = if (!isActive) 0 else controllerContext.partitionLeadershipInfo.size
+
+    topicsToDeleteCount = if (!isActive) 0 else {
+      (controllerContext.topicsToBeDeleted -- controllerContext.topicsIneligibleForDeletion).size
+    }
+
+    replicasToDeleteCount = if (!isActive) 0 else controllerContext.topicsToBeDeleted.map { topic =>
+      // For each enqueued topic, count the number of eligible replicas that are not yet deleted
+      controllerContext.replicasForTopic(topic).count { replica =>
+        controllerContext.replicaState(replica) != ReplicaDeletionSuccessful &&
+        controllerContext.replicaState(replica) != ReplicaDeletionIneligible
```

Comment:
Hmm.. Why do we exclude replicas that are currently ineligible? I think eventually they still need to get deleted.

Reply:
The idea is that the two counts (replicasToDelete and ineligibleReplicasToDelete) are mutually exclusive so they can be combined to see the total number of pending replica deletes. This was a suggestion in the KIP discussion. Alternatively, we could have replicasToDeleteCount represent _all_ pending replica deletes regardless of eligibility.

Reply:
Hmm, I see. I guess that's reasonable. It seems to make the naming a little less clear though. For example, `TopicsToDeleteCount` really means something like `EligibleTopicsToDeleteCount`. I think I slightly favor letting `TopicsToDeleteCount` just represent the count of topics that are being deleted. Similarly for replicas.

Reply:
Yea, i actually kind of like prefer that as well. Updated.